browser/IgBlazorSamples.Gulp/tasks/gulp-samples.js

```diff
@@ -122,70 +122,72 @@ function getSamples(cb) {
-        // console.log("get " + sampleFolder + '/');
+            let sampleFolder = Transformer.getRelative(sample.dirname);
+            // console.log("get " + sampleFolder + '/' + sample.basename);
+            // console.log("get " + sampleFolder + '/');
 
-        let sampleFiles = [];
-        gulp.src([
+            let sampleFiles = [];
+            gulp.src([
                 sampleFolder + "/Pages/*",
                 sampleFolder + "/Components/*",
                 sampleFolder + "/Services/*",
                 sampleFolder + "/*.csproj",
                 sampleFolder + "/wwwroot/*.js",
                 sampleFolder + "/wwwroot/*.css",
-             // sampleFolder + "/wwwroot/*",
-          '!' + sampleFolder + "/wwwroot/index.html",
-          '!' + sampleFolder + "/wwwroot/index.css",
-          '!' + sampleFolder + "/Pages/_*.razor",
-        //   '!' + sampleFolder + "/Pages/DataGridBindingLiveData.razor",
-          '!' + sampleFolder + "/Pages/*.g.cs",
-          '!' + sampleFolder + "/obj/**",
-          '!' + sampleFolder + "/obj/*.*",
-          '!' + sampleFolder + "/bin/**",
-          '!' + sampleFolder + "/bin/*.*",])
-        .pipe(flatten({ "includeParents": -1 }))
-        .pipe(es.map(function(file, fileCallback) {
-            let fileDir = Transformer.getRelative(file.dirname);
-            sampleFiles.push(fileDir + "/" + file.basename);
-            // console.log("get file " + fileDir + "/" + file.basename);
-            fileCallback(null, file);
+                // sampleFolder + "/wwwroot/*",
+                '!' + sampleFolder + "/wwwroot/index.html",
+                //   '!' + sampleFolder + "/wwwroot/index.css",
+                '!' + sampleFolder + "/Pages/_*.razor",
+                //   '!' + sampleFolder + "/Pages/DataGridBindingLiveData.razor",
+                '!' + sampleFolder + "/Pages/*.g.cs",
+                '!' + sampleFolder + "/obj/**",
+                '!' + sampleFolder + "/obj/*.*",
+                '!' + sampleFolder + "/bin/**",
+                '!' + sampleFolder + "/bin/*.*",])
+                .pipe(flatten({ "includeParents": -1 }))
+                .pipe(es.map(function (file, fileCallback) {
+                    let fileDir = Transformer.getRelative(file.dirname);
+
+                    let sampleFilePush = fileDir + "/" + file.basename;                    
+                    sampleFiles.push(fileDir + "/" + file.basename);
+                    // console.log("get file " + fileDir + "/" + file.basename);
+                    fileCallback(null, file);
+                }))
+                .on("end", function () {
+                    // log(sampleFolder);
+
+                    let sampleInfo = Transformer.getSampleInfo(sample, sampleFiles);
+                    if (sampleInfo !== null) {
+                        samples.push(sampleInfo);
+                    }
+                    sampleCallback(null, sample);
+                });
```

Comment:
moving ".on("end", function () " here will cause creating SampleInfo for each file in a sample directory rather than one SampleInfo for each sample directory. Was there a reason why you have to move ".on("end", function () " ?